gensim/parsing/preprocessing.py

```diff
@@ -89,8 +92,32 @@ def remove_stopwords(s):
         u'Better late never, better late.'
 
     """
+    if stopwords is None:
+        stopwords = STOPWORDS
     s = utils.to_unicode(s)
-    return " ".join(w for w in s.split() if w not in STOPWORDS)
+    return " ".join(w for w in s.split() if w not in stopwords)
+
+
+def remove_stopword_tokens(tokens, stopwords=None):
```

Comment:
Looks like a duplicate of `remove_stopwords` – do we need both?

Reply:
Yes both the functions are quite similar. The only difference is that `remove_stopwords()` take a string as an argument and also return string while the `remove_stopword_tokens()` take a list of tokens as argument and return a list.  (same thing for `strip_short()` and `remove_short_tokens()`)

`remove_stopwords()` is what widely used across the gensim package. But `remove_stopword_tokens()` is used only in the `textcorpus` module as it first tokenizes the text and then using preprocessing filters.

Though we can remove these duplicate functions by doing some minor changes (doing the preprocessing filters first and then use the tokenizer) in the `Textcorpus` class.

Reply:
Ah sorry, I completely missed that difference.

It's probably OK then, if the signatures differ. Does it make sense to implement one in terms of the other? To keep it at least a little DRY.

Reply:
>Does it make sense to implement one in terms of the other? 

Yes, we can do that to minimize the repetition.